pkg/noderesourcetopology/pluginhelpers.go

```diff
@@ -139,12 +145,59 @@ func makePodByResourceListWithManyContainers(resources *v1.ResourceList, contain
 func extractResources(zone topologyv1alpha1.Zone) v1.ResourceList {
 	res := make(v1.ResourceList)
 	for _, resInfo := range zone.Resources {
-		klog.V(5).InfoS("Extract resources for zone", "resName", resInfo.Name, "resAvailable", resInfo.Available)
 		res[v1.ResourceName(resInfo.Name)] = resInfo.Available
 	}
 	return res
 }
 
+func logNumaNodes(desc, nodeName string, nodes NUMANodeList) {
+	if !klog.V(6).Enabled() {
+		// avoid even the trivial range over nodes
+		return
+	}
+	for _, numaNode := range nodes {
+		numaLogKey := fmt.Sprintf("%s/node-%d", nodeName, numaNode.NUMAID)
+		klog.V(6).InfoS(desc, resourceListToLoggable(numaLogKey, numaNode.Resources)...)
+	}
+}
+
+func resourceListToLoggable(logKey string, resources v1.ResourceList) []interface{} {
+	// count pairs + logKey pair
+	items := make([]interface{}, (len(resources)+1)*2)
+	items[0] = "logKey"
+	items[1] = logKey
+
+	idx := 0
```

Comment:
how about just get rid of the index thing:

```go
func resourceListToLoggable(logKey string, resources v1.ResourceList) []interface{} {
	// count pairs + logKey pair
	items := make([]interface{}, 0)
	items = append(items, "logKey")
	items = append(items, logKey)

	resNames := make([]string, 0)
	for resName := range resources {
		resNames = append(resNames, string(resName))
	}
	sort.Strings(resNames)

	for _, resName := range resNames {
		qty := resources[v1.ResourceName(resName)]
		items = append(items, resName)
		resVal, _ := qty.AsInt64()
		if needsHumanization(resName) {
			items = append(items, humanize.IBytes(uint64(resVal)))
		} else {
			// fmt.Sprintf is an overkill, let's use something lighter
			items = append(items, strconv.FormatInt(resVal, 10))
		}
	}
	return items
}
```

Reply:
this looks nicer indeed. I was doing some premature optimization towards speed, but I don't have any data to back this choice (hence premature), so let's start readable and let's optimize later where/if there is a clear need. Thanks!